Disable Debug Assertion for Parsing NPM Person objects

mjbvz opened the pull request:
**Defect**
While debugging, Npm `Person` ends up throwing thousands and thousands of exceptions because the Json data it is receiving from NPM is invalid according to its current check. It expects an object, but a person in a package.json may also be a string: https://docs.npmjs.com/files/package.json#people-fields-author-contributors

**Fix**
* Moved the parsing logic for `Person` to a factory named `CreateFromJsonSource`. This should make it clear to callers what this function does.
* For parsing, also accept just regular json strings as input. The new factory does not parse this field currently but uses it as the name field for now. Parsing may be added later.
* Added some unit tests to pin the existing behavior and test the new string behavior.


A whitespace only change to InteractiveWindow.cs is included in this PR. Git automatically created this change and I cannot undo it without manually editing the .gitattributes file. I think we just need to commit this file again so that git is no longer confused about the line endings in this file.

msftclas commented:
Hi __@mjbvz__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Matt Bierner). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla.microsoft.com.
    </p>

TTYL, MSBOT;

mjbvz commented on Nodejs/Product/Npm/SPI/Person.cs:
```diff
@@ -29,68 +29,99 @@ internal class Person : IPerson {
-                    if (group.Success) {
-                        Name = group.Value;
+                    if (@group.Success) {
+                        name = @group.Value;
```
Weird... I was wondering about those names but thought it was intentional. Reshaper's extract messed something up here. Will undo this.

mousetraps commented on Common/Tests/Utilities.UI/UI/InteractiveWindow.cs:
```diff
@@ -1,310 +1,310 @@
-﻿/* ****************************************************************************
```
You should be able to remove this change now that #709 has been merged

mousetraps commented on Nodejs/Product/Npm/SPI/Person.cs:
```diff
@@ -29,68 +29,99 @@ internal class Person : IPerson {
-        "\"url\":\\s*\"(?<url>[^<]+?)\"",
-        RegexOptions.Singleline);
-    
+        private static readonly Regex ObjectPersonRegex = new Regex(
```
nit: `PersonObjectRegex` and `StringPersonRegex` would be clearer

mjbvz replied:
Just removed `StringPersonRegex` since it was not really being used

mousetraps commented on Nodejs/Product/Npm/SPI/Person.cs:
```diff
@@ -29,68 +29,99 @@ internal class Person : IPerson {
+            if (stringPerson != null)
+                return stringPerson;
+
+            return new Person(source);
```
`return TryCreatePersonFromObject(source) ?? TryCreatePersonFromString(source) ?? new Person(source);`

mousetraps commented on Nodejs/Product/ProjectWizard/NodejsPackageParametersExtension.cs:
```diff
@@ -19,7 +19,7 @@
 using System.Text.RegularExpressions;
```
remove this file from changeset too - again, I think #730 should make things easier :smiley:

mousetraps commented on Nodejs/Tests/NpmTests/PersonTests.cs:
```diff
@@ -0,0 +1,96 @@
+namespace NpmTests {
+    [TestClass]
+    public class PersonTests {
+        [TestMethod, TestCategory("UnitTest")]
```
TestCategory("npm-integration") if we want to start working it in?

mousetraps commented on Nodejs/Product/Npm/SPI/Person.cs:
```diff
@@ -29,68 +29,99 @@ internal class Person : IPerson {
+                var group = match.Groups["name"];
+                if (group.Success)
+                    return new Person(group.Value);
             }
```
If we are not parsing the string, then why are we sending it through a regex?

mjbvz commented:
Ok, I removed the string person regex stuff for now, addressed the other comments, and have synced with master to get rid of any line ending changes.

mjbvz commented:
@mousetraps can you please take another quick look at this review. This currently makes debugging a pain if you ever accidentally trigger the npm download, plus the assertion is not even correct to being with.

mousetraps commented:
test categories still need to be updated but :+1: after that!
